dataset/src/main/scala/frameless/Job.scala

```diff
@@ -22,8 +22,17 @@ sealed abstract class Job[A](implicit sc: SparkContext) { self =>
       }
     }
   }
+
+  def map[B](fn: A => B): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run())
+  }
+
+  def flatMap[B](fn: A => Job[B]): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run()).run()
+  }
```

Comment:
I think adding the `dataset` project as a dependency to `cats` would be fine. (If someone complains that they want the `cats` part in isolation, it would be easy to split after the fact).